# Rotation1 Group4
## Introduction
This project is about LIFE4136 being carried out in Group4 of Rotation1. This research project is about the strain of a microorganism. We need to study which places or parts of the genome have been modified and where what changes have occurred. Our data includes short-read (Illumina data) and long-read data for some samples, but the majority of the samples only have long-read data. We want a collection of these genomes to determine what has happened to them.
## Installation
A conda environment (named 'myenv' was already created during the environment setup). If not created yet, we can create a new conda environment using the following command:

```bash
conda create -n bioinfo_env python=3.8
conda activate bioinfo_env conda create -n bioinfo_env python=3.8
conda activate bioinfo_env
```
Install required tools using commands:
```bash
conda install -c bioconda unicycler=0.5.0
conda install -c bioconda nanoplot=1.42.0
conda install -c bioconda quast=5.2.0
conda install -c bioconda prokka=1.14.6
conda install -c bioconda minimap2=2.28
conda install bioconda：：bwa
conda install bioconda：：samtools
```
## Data

Data Stages Table

| Stage                        | Input Data                                                                 | Output Data                                                                 |
|------------------------------|---------------------------------------------------------------------------|----------------------------------------------------------------------------|
| **Short Read QC**            | - `R1merged_08.fastq`<br>- `R2merged_08.fastq`                             | - `qc_R1_shortread_b8-2282057.err`<br>-<br>- `qc_R1_shortread_b8-2282057.out`<br> |
| **Long Read QC**             | - `pf_merged_barcode08.fastq.gz`                                          | - `passnanoplot_results`                                                   |
| **High-Depth Hybrid Assembly** | - `highdepth_hybrid-2163525.err`<br>- `highdepth_hybrid-2163525.out`       | - `001_spades_graph_k015.gfa`<br>- `002_depth_filter.gfa`<br>- `003_overlaps_removed.gfa`<br>- `004_long_read_assembly.gfa`<br>- `005_bridges_applied.gfa`<br>- `006_final_clean.gfa`<br>- `assembly.gfa`<br>- `highdepth_assembly.fasta`<br>- `highdepth_assembly.fasta.fai`<br>- `unicycler.log` |
| **Low-Depth Hybrid Assembly** | - `lowdepth_hybrid-2163529.out`                                           | - `001_spades_graph_k015.gfa`<br>- `002_depth_filter.gfa`<br>- `003_overlaps_removed.gfa`<br>- `004_long_read_assembly.gfa`<br>- `005_bridges_applied.gfa`<br>- `006_final_clean.gfa`<br>- `assembly.gfa`<br>- `lowdepth_assembly.fasta`<br>- `lowdepth_hybrid-2163529.err`<br>- `unicycler.log` |
| **Pass Read Assembly**       | - `pf_merged_barcode08.fastq.gz`                                          | - `slurm-pf_assembly-2145892.err`<br>- `slurm-pf_assembly-2145894.err`<br>- `slurm-pf_assembly-2145892.out`<br>- `slurm-pf_assembly-2145894.out`<br>- `unicycler.log`<br>- `pf_assembly.fasta`<br>- `pf_longread_assembly.sh` |
| **Annotation**               | - `hvolcanii_ref.fna`<br>- `genes.gff`<br>- `genes.txt`                   | - `genome_annotation`<br>- `gffread`<br>- `long_read`<br>- `pilonpolish`<br>- `quast`<br>- `short_read` |
| **Quast Evaluation**         | - `pass_assembly.fasta`<br>- `pf_assembly.fasta`                           | - `aligned_stats`<br>- `basic_stats`<br>- `contigs_reports`<br>- `compare_assemblies_mergep4ref.sh`<br>- `compare_assembly_methods.sh`<br>- `genome_stats`<br>- `icarus.html`<br>- `report.html`<br>- `report.pdf`<br>- `report.tex`<br>- `report.tsv`<br>- `transposed_report.txt`<br>- `transposed_report.tex`<br>- `transposed_report.tsv` |
| **BWA Alignment**            | - `sorted_shortreads_mergeref.bam`                                        | - `cp_p4inc_merge_ref.fasta.bwt`<br>- `cp_p4inc_merge_ref.fasta.pac`<br>- `cp_p4inc_merge_ref.fasta.sa`<br>- `index_ref-2185130.err`<br>- `index_ref-2185130.out`<br>- `make_sam-2185872.err`<br>- `make_sam-2185872.out`<br>- `p4_make_sam-2188671.err`<br>- `p4_make_sam-2188671.out`<br>- `p4inc_bwa_bam_bai-2188679.err`<br>- `p4inc_bwa_bam_bai-2188679.out` |

### Notes:
- **Input Data**: The input data includes FASTQ files (`R1merged_08.fastq`, `R2merged_08.fastq`, `pf_merged_barcode08.fastq.gz`), reference sequences (`hvolcanii_ref.fna`), and annotation files (`genes.gff`, `genes.txt`).
- **Output Data**: The output data includes various assembly graphs (`.gfa`), filtered and cleaned assemblies, FASTA files, logs from the assembly process, evaluation reports, and statistics generated by Quast.

This table serves as a guide to understand the data through each stage of the analysis pipeline.

## Code
### 1. Merging files

| Script Name                                      | Description                     |
|--------------------------------------------------|----------------------------------|
| [longreadmerge.sh](/merger/longreadmerge.sh) | Merge long-read FASTQ files    |
| [merger.sh](/merger/merger.sh)   | General long-read merger        |
| [short_read_R1_merger.sh](/merger/short_read_R1_merger.sh) | Merge R1 short-read files       |
| [short_read_R2_merger.sh](/merger/short_read_R2_merger.sh) | Merge R2 short-read files       |

### 2. QC

Using the Nanoplot package

| Script Name                                      | Description                     |
|--------------------------------------------------|----------------------------------|
| [R1_qc.sh](/merger/R1_qc.sh)     | R1 short-read QC                |
| [R2_qc.sh](/merger/R2_qc.sh)     | R2 short-read QC                |
| [passmerger.sh](/merger/passmerger.sh) | Merge QC-passed long-read files |
| [failmerger.sh](/merger/failmerger.sh) | Merge QC-failed long-read files |

### 3. Genome assembly

Using the Unicylcer package

| Script Name                  | Description                              |
|------------------------------|------------------------------------------|
| [highdepth_hybrid_assembly.sh](/barcode8/assembly/highdepth_hybrid_assembly.sh) | High-depth hybrid assembly script        |
| [lowdepth_hybrid_assembly.sh](/barcode8/assembly/lowdepth_hybrid_assembly.sh)  | Low-depth hybrid assembly script         |
| [pass_longread_assembly.sh](/barcode8/assembly/pass_longread_assembly.sh)    | Assemble passing long-reads post-QC      |
| [pf_longread_assembly.sh](/barcode8/assembly/pf_longread_assembly.sh)      | Assemble partially filtered long-reads   |
| [shortread_assembly.sh](/barcode8/assembly/shortread_assembly.sh)        | Short-read assembly script               |

### 4. Quast

Using the Quast package

| Script Name                                      | Description                     |
|--------------------------------------------------|----------------------------------|
| [quast.sh](/barcode8/quast.sh)     | combined quast                |
| [compare_assemblies_mergep4ref.sh](/barcode8/compare_assemblies_mergep4ref.sh )     | compare assembly methods   |
| [longread_quast.sh](/barcode8/longread_quast.sh)     |longread_eval                 |
| [shortread_quast.sh](/barcode8/shortread_quast.sh) | shortread_eval |
| [hybrid_quast.sh](/barcode8/hybrid_quast.sh) | hybrid_eval |

### 5. Genome annotation

Using the Prokka package

| Script Name                                      | Description                     |
|--------------------------------------------------|----------------------------------|
| [compliant_hybrid_prokka.sh](/barcode8/compliant_hybrid_prokka.sh)     |  compliant hybrid               |
| [highdepth_hybrid_prokka.sh](/barcode8/highdepth_hybrid_prokka.sh)     |  highdepth hybrid               |
| [lowdepth_hybrid_prokka.sh](/barcode8/lowdepth_hybrid_prokka.sh)     | lowdepth hybrid   |
| [prokka_pass_longreads.sh](/barcode8/prokka_pass_longreads.sh)     |pass longreads        |
| [prokka.sh](/barcode8/prokka.sh) | short reads |

Annotating references

| Script Name                                      | Description                     |
|--------------------------------------------------|----------------------------------|
| [merge_ref.sh](/barcode8/merge_ref.sh)     |  merge the references               |
| [merge_ann_ref.sh](/barcode8/merge_ann_ref.sh)     |     annotationg the merge references           |

Using genovi package

[complete_highdepth_hybrid_genovi.sh](/barcode8/complete_highdepth_hybrid_genovi.sh)


### 6. IGV

Using this website to download IGV (https://igv.org/doc/desktop/#DownloadPage/)

Using the bwa package

Using the samtools package

| Script Name                                      | Description                     |
|--------------------------------------------------|----------------------------------|
| [bam_bai_maker.sh](/barcode8/bam_bai_maker.sh)     |  creating sorted shortread .BAM.bai file              |
| [p4inc_bam_bai_maker.sh](/barcode8/p4inc_bam_bai_maker.sh)     | Creating shortread .BAM.bai file              |
| [filter_end.sh](/barcode8/filter_end.sh)     | Filter for end region            |

| Script Name                                      | Description                     |
|--------------------------------------------------|----------------------------------|
| [bam_bai_converter.sh](/barcode8/bam_bai_converter.sh)     |  creating sorted longread .BAM.bai file              |
| [p4inc_bam_bai_converter.sh](/barcode8/p4inc_bam_bai_converter.sh)     | Creating longread .BAM.bai file              |


## References
